File: parse_html.js
const fs = require('fs');
const jsdom = require("jsdom");
const { JSDOM } = jsdom;
const html = fs.readFileSync('app/src/main/assets/index.html', 'utf8');
const dom = new JSDOM(html);
console.log("ai-cups:", dom.window.document.getElementById('ai-cups'));
console.log("body children count:", dom.window.document.body.children.length);

File: parse_html.py
from html.parser import HTMLParser
import sys

class MyHTMLParser(HTMLParser):
    def handle_starttag(self, tag, attrs):
        for attr in attrs:
            if attr[0] == 'id' and attr[1] == 'ai-cups':
                print(f"Found ai-cups at line {self.getpos()[0]}")

parser = MyHTMLParser()
with open('app/src/main/assets/index.html', 'r') as f:
    parser.feed(f.read())
